src/components/bar-chart/bar-chart.jsx

```diff
@@ -0,0 +1,123 @@
+import { useEffect, useState } from 'react';
+import {
+    BarChart as BarChartWrapper, Bar, XAxis, YAxis, CartesianGrid, Tooltip, Legend, ResponsiveContainer
+} from 'recharts';
+import ChartLegendContent from './chart-legend-content';
+import ChartTooltipContent from './chart-tooltip-content';
+
+const BarChart = ({
+    data,
+    dataKeys = [],
+    colors = [],
+    layout = "horizontal", // horizontal, vertical
+    stacked = false,
+    showXAxis = true,
+    showYAxis = true,
+    showTooltip = true,
+    tooltipIndicator = 'dot', // dot, line, dashed
+    tooltipLabelKey,
+    showLegend = false,
+    showCartesianGrid = true,
+    tickFormatter,
+    xAxisDataKey,
+    yAxisDataKey,
+    xAxisFontSize = 'sm', // sm, md, lg
+    xAxisFontColor = "#4B5563",
+    yAxisFontColor = "#4B5563",
+    chartWidth = 350,
+    chartHeight = 200,
+    borderRadius = 8,
+}) => {
+
+    const [width, setWidth] = useState(chartWidth);
+    const [height, setHeight] = useState(chartHeight);
+
+    useEffect(() => {
+        setWidth(chartWidth);
+        setHeight(chartHeight);
+    }, [chartWidth, chartHeight]);
+
+    const fontSizeMap = {
+        sm: '12px',
+        md: '14px',
+        lg: '16px'
```

Comment:
can we use text-xs, text-sm, text-md

Reply:
No, because I need set font size to tick property for XAxis component like this.
` <XAxis
                        dataKey={xAxisDataKey}
                        tickLine={false}
                        axisLine={false}
                        tickMargin={8}
                        tickFormatter={tickFormatter}
                        tick={{ fontSize: fontSizeVariant, fill: xAxisFontColor }}
                    />
`